lambdas/preview/index.py

```diff
@@ -163,13 +168,61 @@ def extract_vcf(file_):
                 break
 
     # pylint: disable=bad-continuation
-    html = ("<div>"
+    html = ('<div>'
                 "<div data-type='meta'>{}</div>"
                 "<div data-type='header'>{}</div>"
                 "<div data-type='data'>{}</div>"
-            "</div>").format(
-                ''.join(meta) + '<br />',
-                ''.join(header) + '<br />',
+            '</div>').format(
+                ''.join(meta),
+                ''.join(header),
                 ''.join(data))
 
     return html, {}
+
+def extract_txt(file_):
+    """
+    Display first and last lines of file.
+    Uses UNIX utils for speed and convenience (no easy way to tail in Python).
+
+    Args:
+        file_ - file-like
+
+    Throws:
+        CalledProcessError
+    """
+    available = 0
+    ellipsis = False
+    # find out how many lines <= MAX_LINES are available
+    # we need to know this so that head and tail don't overlap
+    # loop finishes in constant time
+    with open(file_.name, 'rb') as txt:
+        for index, line in enumerate(txt, start=1):
+            available = index
+            if available > MAX_LINES:
+                ellipsis = True
+                break
+    # if maxlines == 0, everything still works
+    available = min(available, MAX_LINES)
+    # pylint should never flag for 3.6 but it does
+    headlines = int(available/2) # pylint: disable=old-division
+    taillines = available - headlines
+
+    head = subprocess.check_output(['head', file_.name, '-n', str(headlines)])
+    tail = subprocess.check_output(['tail', file_.name, '-n', str(taillines)])
```

Comment:
You can use SEEK_END to scan from the end of the file.
```
    f.seek(-2, os.SEEK_END)     # Jump to the second last byte.
    while f.read(1) != b"\n":   # Until EOL is found...
        f.seek(-2, os.SEEK_CUR) # ...jump back the read byte plus one more.
    last = f.readline()    
```
And for head, I think you don't need anything special, you can just open the file and read the first n lines as normal.

Reply:
wanted to avoid but looks easy enough; let me see if i can use this